### DISCUSSION TITLE
Cannot update a B2B offer using patchListingsItem

### DISCUSSION THREAD

**Discussion by domeglic** [amzn/selling-partner-api-models#4390] [General]
Several of our products are not shown because the price is lower than the B2B price so we get the error that the minimal price is higher. This B2B price is a new thing based on what I read but I cannot update this price using the API.

Currently we send a purchasable_offer and I have modifed it to add `audience:ALL` and add another offer with the same price and `audience:B2B`.

When I look at the item using `getListingsItem` I can see my updated `purchasable_offer` but the `offers` still has the old B2B offer. I also tried with a lower price for b2b and to use `putListingsItem` by sending all data again. Nothing is changed.

What am I missing? Do I have to delete and recreate them?

**Comment by jawheral** [comment]
Hi @domeglic,

If `getListingsItem` returned the `purchasable_offer` as you expect it, then I suggest that you wait some time. Updates can be delayed and take some time to reflect.

If you still experience issues, then please contact [Developer Support](https://developer.amazonservices.com/support) while providing all the relevant data.

Best,
Jawher
Selling Partner Developer Services

**Comment by domeglic** [comment]
Now the b2b offer finally changed but it still says there is a price problem. We never set a minimal price and I cannot see it in the data returned by getListingItem, but in the Amazon manage inventory I can see a minimum price. It seems like the API is not returning all the data that it should.

**Reply by jawheral** [reply]
Hi @domeglic,

Can you share the exact error message for the price problem? 
The Listings API should return the minimum price as part of the `purchasable_offer` when available in the `minimum_seller_allowed_price` attribute.

If you need more assistance, then please contact [Developer Support](https://developer.amazonservices.com/support) while providing all the relevant data such as SellerId, SKU and marketplace.

Best,
Jawher
Selling Partner Developer Services

**Comment by domeglic** [comment]
The error in the issues is:
```json
{
            "code": "18639",
            "message": "The selling price was identified as a potential low pricing error. Please review and confirm or update your price.",
            "severity": "ERROR",
            "categories": [
                "INVALID_PRICE"
            ],
            "enforcements": {
                "actions": [
                    {
                        "action": "LISTING_SUPPRESSED"
                    }
                ],
                "exemption": {
                    "status": "NOT_EXEMPT"
                }
            }
        }
```

The listings didn't return the minimum price, just `our_price`. And offers have no minimum price:
```json
"offers": [{
            "marketplaceId": "A1PA6795UKMFR9",
            "offerType": "B2C",
            "price": {
                "currency": "EUR",
                "currencyCode": "EUR",
                "amount": "91.92"
            },
            "audience": {
                "value": "ALL",
                "displayName": "Bei Amazon verkaufen"
            }
        },
        {
            "marketplaceId": "A1PA6795UKMFR9",
            "offerType": "B2B",
            "price": {
                "currency": "EUR",
                "currencyCode": "EUR",
                "amount": "91.92"
            },
            "audience": {
                "value": "B2B",
                "displayName": "Amazon Business (B2B)"
            }
        }
]
```

I have now also tried to set a minimum price using a patchListingsItem and set the schedule with a date in the past to try and remove it. But nothing changed. The error remains and the Amazon admin panel still shows a minimum price of 212€.

**Reply by jawheral** [reply]
Does the `purchasable_offer` attribute when requesting `getListingsItem` by including `attributes` in the `includedData` parameter contain the minimum price?

**Comment by domeglic** [comment]
Before I did anything it did not. The 212 was the minimum price shown in the admin panel but not in the data recieved by the getListingItem. Now that I have sent the minimum price it did change eventually.

But this is still a problem because we do not use minimum prices ever but somethow listings have them set. 

Just today we have a brand new offer that was created using this payload:

```json
{
   "condition_type": [
      {
         "value:" "new_new",
        "marketplace_id": "A1PA6795UKMFR9"
      },
   ],
   "purchasable_offer": [
      {
         "currency": "EUR",
         "our_price": [
            {
               "schedule": [
                  {
                     "value_with_tax": 11.98
                  }
               ]
            }
         ],
         "marketplace_id": "A1PA6795UKMFR9",
      }
   ],
   "fulfillment_availability": [
      {
         "fulfillment_channel_code": "DEFAULT",
         "quantity": 1991,
         "marketplace_id": "A1PA6795UKMFR9"
      }
   ],
   "merchant_shipping_group": [
      {
         "value": "xxx",
         "marketplace_id": "A1PA6795UKMFR9"
      }
   ]
}
```

But it somehow has a minimum price set that is 11.97, so 1 cent less. And it has a b2b offer that is 15.41. I don't know if this is being done by some Amazon service that was enabled, but we don't have other API services.

I don't know what happens to this new product if we change the price, it will likely end up with the same problem since we don't send the minimum price.

**Reply by jawheral** [reply]
@domeglic 
This doesn't seem to be an API related concern. I understand you are creating offers for products already existing on Amazon Catalog. 
I would suggest to check if there are any pricing restrictions on the ASINs for which you are creating offers or check if there any pricing automation settings setup in your account.
You can also contact [Seller Support](https://sellercentral.amazon.com/help/hub/support) to rule out such issues.

Best,
Jawher
Selling Partner Developer Services
